plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationPollingService.kt

```diff
@@ -0,0 +1,161 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.notifications
+
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.intellij.ide.util.RunOnceUtil
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.Service
+import com.intellij.util.Alarm
+import com.intellij.util.AlarmFactory
+import com.intellij.util.io.HttpRequests
+import software.aws.toolkits.core.utils.RemoteResolveParser
+import software.aws.toolkits.core.utils.RemoteResource
+import software.aws.toolkits.core.utils.error
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.jetbrains.core.DefaultRemoteResourceResolverProvider
+import software.aws.toolkits.jetbrains.core.RemoteResourceResolverProvider
+import java.io.InputStream
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.time.Duration
+
+private const val NOTIFICATION_ENDPOINT = "https://idetoolkits-hostedfiles.amazonaws.com/Notifications/JetBrains/1.json" // TODO: Replace with actual endpoint
+private const val MAX_RETRIES = 3
+private const val RETRY_DELAY_MS = 1000L
+
+interface NotificationPollingService {
+    fun startPolling()
+    fun addObserver(observer: (Path) -> Unit)
+    fun dispose()
+}
+
+object NotificationFileValidator : RemoteResolveParser {
+    override fun canBeParsed(data: InputStream): Boolean {
+        return try {
+            NotificationMapperUtil.mapper.readValue<NotificationsList>(data)
+            true
+        } catch (e: Exception) {
+            false
+        }
+    }
+}
+
+@Service(Service.Level.APP)
+class NotificationPollingServiceImpl :
+    NotificationPollingService,
+    PersistentStateComponent<NotificationPollingServiceImpl.State>,
```

Comment:
We should probably have a separate persistent state class that keeps track of the etag and get its instance here